src/Analyzers/CSharp/Tests/UseIsNullCheck/UseIsNullCheckForReferenceEqualsTests.cs

```diff
@@ -500,5 +501,67 @@ void M(string s2)
     }
 }");
         }
+
+        [WorkItem(23581, "https://github.com/dotnet/roslyn/issues/47972")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestValueParameterTypeIsBaseConstraintGeneric()
+        {
+            await TestInRegularAndScript1Async(
+@"using System;
+
+class C
+{
+    public static void NotNull<T>(T value) where T:C
+    {
+        if ({|FixAllInDocument:ReferenceEquals|}(value, null))
+        {
+            return;
+        }
+    }
+}",
+@"using System;
+
+class C
+{
+    public static void NotNull<T>(T value) where T:C
+    {
+        if (value is null)
+        {
+            return;
+        }
+    }
+}", new TestParameters(parseOptions: CSharp7));
+        }
+
+        [WorkItem(23581, "https://github.com/dotnet/roslyn/issues/47972")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestUnconstrainedGenericInCSharp8()
+        {
+            await TestInRegularAndScript1Async(
+@"using System;
+
+class C
+{
+    public static void NotNull<T>(T value)
+    {
+        if ({|FixAllInDocument:ReferenceEquals|}(value, null))
+        {
+            return;
+        }
+    }
+}",
+@"using System;
+
+class C
+{
+    public static void NotNull<T>(T value)
+    {
+        if (value is null)
+        {
+            return;
+        }
+    }
+}", new TestParameters(parseOptions: CSharp8));
+        }
```

Comment:
can you add an unconstrained test for C#7 as well?

Reply:
That's the one I added `parseOptions: CSharp7` above. I should move them together.